Sources/SwiftGRPCNIOInteroperabilityTests/InteroperabilityTestCases.swift

```diff
@@ -0,0 +1,679 @@
+    let client = Grpc_Testing_TestServiceService_NIOClient(connection: connection)
+    let call = client.emptyCall(Grpc_Testing_Empty())
+
+    try assertEqual(call.response, Grpc_Testing_Empty())
+    try assertEqual(call.status.map { $0.code }, .ok)
+  }
+}
+
+/// This test verifies that gRPC requests marked as cacheable use GET verb instead of POST, and
+/// that server sets appropriate cache control headers for the response to be cached by a proxy.
+/// This test requires that the server is behind a caching proxy. Use of current timestamp in the
+/// request prevents accidental cache matches left over from previous tests.
+///
+/// Server features:
+/// - CacheableUnaryCall
+///
+/// Procedure:
+/// 1. Client calls CacheableUnaryCall with SimpleRequest request with payload set to current
+///    timestamp. Timestamp format is irrelevant, and resolution is in nanoseconds. Client adds a
+///    x-user-ip header with value 1.2.3.4 to the request. This is done since some proxys such as
+///    GFE will not cache requests from localhost. Client marks the request as cacheable by
+///    setting the cacheable flag in the request context. Longer term this should be driven by
+///    the method option specified in the proto file itself.
+/// 2. Client calls CacheableUnaryCall again immediately with the same request and configuration
+///    as the previous call.
+///
+/// Client asserts:
+/// - Both calls were successful
+/// - The payload body of both responses is the same.
+class CacheableUnary: InteroperabilityTest {
+  func run(using connection: GRPCClientConnection) throws {
+    let client = Grpc_Testing_TestServiceService_NIOClient(connection: connection)
+
+    var timestamp = DispatchTime.now().rawValue
+    let request = Grpc_Testing_SimpleRequest.withPayload(of: .bytes(of: &timestamp))
+
+    var headers = HTTPHeaders()
+    headers.add(name: "x-user-ip", value: "1.2.3.4")
+    let callOptions = CallOptions(customMetadata: headers, cacheable: true)
+
+    let call1 = client.cacheableUnaryCall(request, callOptions: callOptions)
+    let call2 = client.cacheableUnaryCall(request, callOptions: callOptions)
+
+    try assertEqual(call1.response.map { $0.payload }, call2.response.map { $0.payload })
+    try assertEqual(call1.status.map { $0.code }, .ok)
+    try assertEqual(call2.status.map { $0.code }, .ok)
+  }
+}
+
+/// This test verifies unary calls succeed in sending messages, and touches on flow control (even
+/// if compression is enabled on the channel).
+///
+/// Server features:
+/// - UnaryCall
+///
+/// Procedure:
+/// 1. Client calls UnaryCall with:
+///    ```
+///    {
+///        response_size: 314159
+///        payload:{
+///            body: 271828 bytes of zeros
+///        }
+///    }
+///    ```
+///
+/// Client asserts:
+/// - call was successful
+/// - response payload body is 314159 bytes in size
+/// - clients are free to assert that the response payload body contents are zero and comparing
```

Comment:
Should we assert that the response payload is zero, for good measure?

Reply:
Might as well; I think this is recommended in a few places to so I'll update them all.